modules/common/tls/tls.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/openstack-k8s-operators/lib-common/modules/common/secret"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Service contains server-specific TLS secret
+type Service struct {
+	// Server-specific settings
+	SecretName             string `json:"secretName,omitempty"`
+	DisableNonTLSListeners bool   `json:"disableNonTLSListeners,omitempty"`
+}
+
+// Ca contains CA-specific settings, which could be used both by services (to define their own CA certificates)
+// and by clients (to verify the server's certificate)
+type Ca struct {
+	// CA-specific settings
+	CaSecretName string `json:"caSecretName,omitempty"`
+}
+
+// TLS - a generic type, which encapsulates both the service and CA configurations
+type TLS struct {
+	Service Service `json:"service"`
+	Ca      Ca      `json:"ca"`
+}
+
+// NewTLS - initialize and return a TLS struct
+func NewTLS(cr client.Object, clientset *kubernetes.Clientset, scheme *runtime.Scheme, namespace string, service Service, ca Ca) (*TLS, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	crClient, err := client.New(cfg, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, err
+	}
+
+	helperInstance, err := helper.NewHelper(cr, crClient, clientset, scheme, log.Log)
+	if err != nil {
+		return nil, fmt.Errorf("could not create helper instance: %w", err)
+	}
+
+	// Ensure service SecretName exists or return an error
+	if service.SecretName == "" {
+		service.SecretName = "default-tls-secret"
+	} else {
+		_, _, err := secret.GetSecret(context.TODO(), helperInstance, service.SecretName, namespace)
+		if err != nil {
+			return nil, fmt.Errorf("error ensuring secret %s exists: %w", service.SecretName, err)
+		}
+	}
+
+	// Ensure ca CaSecretName exists or return an error
+	if ca.CaSecretName == "" {
+		ca.CaSecretName = "default-ca-secret"
+	} else {
+		_, _, err := secret.GetSecret(context.TODO(), helperInstance, ca.CaSecretName, namespace)
+		if err != nil {
+			return nil, fmt.Errorf("error ensuring secret %s exists: %w", ca.CaSecretName, err)
+		}
+	}
+
+	return &TLS{
+		Service: service,
+		Ca:      ca,
+	}, nil
+}
+
+// CreateVolumeMounts - add volume mount for TLS certificate and CA certificates, this counts on openstack-operator providing CA certs with unique names
+func CreateVolumeMounts(tls *Service, ca *Ca) []corev1.VolumeMount {
+	var volumeMounts []corev1.VolumeMount
+
+	if tls.SecretName != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "tls-certs",
+			MountPath: "/var/tls-certs",
+			ReadOnly:  true,
+		})
+	}
+
+	if ca.CaSecretName != "" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "ca-certs",
+			MountPath: "/var/ca-certs",
```

Comment:
lets put them in `/var/lib/config-data/ca-certificates`, thats where I set to expect them in https://github.com/openstack-k8s-operators/tcib/pull/82